packages/styled-token/src/index.js

```diff
@@ -1,38 +1,63 @@
 import { stringToPath } from './stringToPath';
 
+// Use a separate cache for each theme object
+const caches = new Map();
```

Comment:
Would a [`WeakMap`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/WeakMap) work better here?

Reply:
TIL

Oh yeah, that would probably be better!

Reply:
Done!